common/lib/capa/capa/responsetypes.py

```diff
@@ -735,6 +801,92 @@ def get_score(self, student_answers):
     def get_answers(self):
         return {self.answer_id: list(self.correct_choices)}
 
+    def get_extended_hints(self, student_answers, new_cmap):
+        """
+        Extract compound and extended hint information from the xml based on the student_answers.
+        The hint information goes into the msg= in new_cmap for display.
+        Each choice in the checkboxgroup can have 2 extended hints, matching the
+        case that the student has or has not selected that choice:
+          <checkboxgroup label="Select the best snack" direction="vertical">
+             <choice correct="true">Donut
+               <choicehint selected="tRuE">A Hint!</choicehint>
```

Comment:
Totally. I figured the right design is that xml attributes should not be case-sensitive .. figuring that once we're editing it in studio, switching between javascript and python and whatever, it could very easily get typed wrong. I'm pretty sure, say correct="tRuE" works that way for multiple choice, analogously.